src/components/dialogs/retry.vue

```diff
@@ -0,0 +1,242 @@
+  @Action toggleRetryModal!: ActionBindingIF
+
+  /** Whether staff payment is valid. */
+  private isStaffPaymentValid = false
+
+  /** The current tab to display. */
+  private currentTab = this.TAB_RETRY_PAYMENT
+
+  /** Whether payment redirection is in progress. */
+  private isLoadingPayment = false
+
+  /** Whether this dialog is visible. */
+  private isVisible = false
+
+  /** Whether this modal should be shown (per store property). */
+  private get showModal (): boolean {
+    return this.$store.getters[RETRY_MODAL_IS_VISIBLE]
+  }
+
+  /** Clears store property to hide this modal. */
+  async hideModal () {
+    this.isLoadingPayment = false
+    await this.toggleRetryModal(false)
+  }
+
+  /** Depending on value, fetches fees and makes this modal visible or hides it. */
+  @Watch('showModal')
+  async onShowModal (val: boolean): Promise<void> {
+    if (val) {
+      // reset tab id
+      this.currentTab = this.TAB_RETRY_PAYMENT
+
+      const params: FetchFeesParams = {
+        filingType: FilingTypes.NM620,
+        jurisdiction: Jurisdictions.BC,
+        priorityRequest: this.getPriorityRequest // same value as originally
+      }
+
+      // only make visible on success, otherwise hide it
+      if (await this.fetchFees(params)) {
+        this.isVisible = true
+      } else {
+        await this.hideModal()
+      }
+    } else {
+      this.isVisible = false
+    }
+  }
+
+  /** Called when staff clicks "Back" button. */
+  goBack () {
+    // disable validation
+    this.$refs.staffPaymentComponent && this.$refs.staffPaymentComponent.setValidation(false)
+    // go to previous tab
+    this.currentTab = this.TAB_RETRY_PAYMENT
+  }
+
+  /** Called when user clicks Continue/Submit button. */
+  private async confirmPayment () {
+    // FUTURE: enable this for staff payment if needed
+    // if (this.isRoleStaff && getFeatureFlag('staff-payment-enabled')) {
+    //   if (this.currentTab === this.TAB_RETRY_PAYMENT) {
+    //     // disable validation
+    //     this.$refs.staffPaymentComponent && this.$refs.staffPaymentComponent.setValidation(false)
+    //     // go to next tab
+    //     this.currentTab = this.TAB_STAFF_PAYMENT
+    //     return
+    //   }
+    //   if (this.currentTab === this.TAB_STAFF_PAYMENT) {
+    //     // enable validation
+    //     this.$refs.staffPaymentComponent && this.$refs.staffPaymentComponent.setValidation(true)
+    //     // if invalid then stop, else continue
+    //     if (!this.isStaffPaymentValid) return
+    //   }
+    // }
+
+    // navigate to the payment portal
+    this.isLoadingPayment = true
+    const { id, token, nrId, action } = this.pendingPayment
+
+    // Save payment to session
+    this.savePendingPaymentToSession(action, this.pendingPayment)
+
+    const baseUrl = getBaseUrl()
+    const redirectUrl = encodeURIComponent(`${baseUrl}/nr/${nrId}/?paymentId=${id}`)
+    this.redirectToPaymentPortal(token, redirectUrl)
+  }
+
+  /** The first pending payment. */
```

Comment:
In testing I had 2 pending payments, which is an invalid case, but I added this comment just to clear things up :)